src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -262,6 +263,25 @@ void ADIOS2IOHandlerImpl::init(
                         "Must be convertible to string type.");
                 }
             }
+
+            if (engineConfig.json().contains(
+                    adios_defaults::str_treat_unsupported_engine_like))
+            {
+                auto maybeEngine = json::asLowerCaseStringDynamic(
+                    engineConfig
+                        [adios_defaults::str_treat_unsupported_engine_like]
+                            .json());
+                if (!maybeEngine.has_value())
```

Comment:
I guess a boolean flag would also do? "e.g. accept_unspported_engine=true"?

Reply:
If true,  can always return ".bp" at realEngineType(). The actual value of the  treat_supported_engine expected to be one of the known adios engines, and  exactly which one is not relavent I think.

Reply:
> I guess a boolean flag would also do? "e.g. accept_unspported_engine=true"?

That will not be enough because there is no "standard" way how an engine should be treated. Especially, most engines differ slightly in how steps should be treated (staging engines require steps, BP5 requires steps with normal Read mode, but cannot deal with steps in ReadRandomAccess, the Campaign engine fails when using steps in any way, other engines are mostly a mixture). 

I don't think that providing one generic implementation for an unknown future engine will work, it seems better to use the existing implementations and let users try which works best.

> If true, can always return ".bp" at realEngineType(). The actual value of the treat_supported_engine expected to be one of the known adios engines, and exactly which one is not relavent I think.

`realEngineType()` will return the Engine type that is actually passed on to ADIOS2. `m_engineType` determines the behavior of the ADIOS2 backend. Especially BP5 behaves slightly different from the other BP engines.
I don't see a reason to put extra work into restricting the options here.

Reply:
Sure. Thanks